lib/hooks/index.js

```diff
@@ -97,25 +58,45 @@ function findModuleVersion(request, parent, load) {
     if (mainScriptDir.indexOf(paths[i]) === 0) {
       // This is unsafe on windows if request contains a namespace
       // TODO: convert path separators in request
-      var pjson = load(path.join(paths[i], request, 'package.json'));
-      return pjson.version;
+      return path.join(paths[i], request);
     }
   }
   return null;
 }
 
+function findModuleVersion(modulePath, load) {
+  return modulePath ? load(path.join(modulePath, 'package.json')).version :
+    process.version;
+}
+
 function activate(agent) {
 
   logger = agent.logger;
 
-  function loadHookAndPatch(loadFn, instrumentation) {
-    assert(instrumentation.module);
-    assert(!instrumentation.hook);
-
+  function loadHookAndPatch(loadFn, instrumentation, moduleRoot, version) {
```

Comment:
My understanding of the module loader is that the second time express@4.0.1 is loaded, the cached version will be used and this cached version will correspond to the module that was originally loaded at a different path. I will put together a small example to confirm this is actually the case.